execution-service/src/main/java/org/ihtsdo/rvf/execution/service/whitelist/AcceptanceGatewayClient.java

```diff
@@ -0,0 +1,86 @@
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.http.*;
+import org.springframework.http.client.ClientHttpRequestExecution;
+import org.springframework.http.client.ClientHttpRequestInterceptor;
+import org.springframework.http.client.ClientHttpResponse;
+import org.springframework.http.converter.json.GsonHttpMessageConverter;
+import org.springframework.web.client.RestTemplate;
+import us.monoid.web.Resty;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.util.List;
+import java.util.Set;
+
+public class AcceptanceGatewayClient {
+    private String acceptanceGatewayServiceUrl;
+    private RestTemplate restTemplate;
+    private HttpHeaders headers;
+    private Resty resty;
+    private static final String ALL_CONTENT_TYPE = "*/*";
+    private static final ParameterizedTypeReference<List<WhitelistItem>> WHITELIST_ITEM_LIST_TYPE_REFERENCE = new ParameterizedTypeReference<List<WhitelistItem>>() {};
+
+    protected static Gson gson;
+    static {
+        GsonBuilder gsonBuilder = new GsonBuilder();
+        gsonBuilder.setPrettyPrinting();
+        gsonBuilder.disableHtmlEscaping();
+        gsonBuilder.excludeFieldsWithoutExposeAnnotation();
+        gson = gsonBuilder.create();
+    }
+
+    public AcceptanceGatewayClient(String acceptanceGatewayServiceUrl, String authToken) throws URISyntaxException, IOException {
+        this.acceptanceGatewayServiceUrl = acceptanceGatewayServiceUrl;
+
+        resty = new Resty(new RestyOverrideAccept(ALL_CONTENT_TYPE));
+        resty.withHeader("Cookie", authToken);
+        resty.withHeader("Connection", "close");
+        resty.authenticate(this.acceptanceGatewayServiceUrl, null,null);
+
+        headers = new HttpHeaders();
+        headers.add("Cookie", authToken);
+        headers.setContentType(MediaType.APPLICATION_JSON);
+        restTemplate = new RestTemplateBuilder()
+                .rootUri(this.acceptanceGatewayServiceUrl)
+                .additionalMessageConverters(new GsonHttpMessageConverter())
+                .errorHandler(new ExpressiveErrorHandler())
+                .build();
+
+        //Add a ClientHttpRequestInterceptor to the RestTemplate to add cookies as required
+        restTemplate.getInterceptors().add(new ClientHttpRequestInterceptor(){
+            @Override
+            public ClientHttpResponse intercept(HttpRequest request, byte[] body, ClientHttpRequestExecution execution) throws IOException {
+                request.getHeaders().addAll(headers);
+                return execution.execute(request, body);
+            }
+        });
+    }
+
+    public static AcceptanceGatewayClient createClient(String acceptanceGatewayServiceUrl, String imsUrl, String username, String password) {
+        IMSRestClient imsClient = new IMSRestClient(imsUrl);
+        String token = null;
+        try {
+            token = imsClient.loginForceNewSession(username, password);
```

Comment:
What does force new session do? Will this invalidate other existing sessions for this user? This could cause problems if there are multiple RVF instances..